File: src/laohuangcode/providers.py
"""Built-in model provider presets."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class Provider:
    name: str
    default_model: str | None
    base_url: str | None
    api_key_env: str


_PROVIDERS = {
    "custom": Provider(
        name="custom",
        default_model=None,
        base_url=None,
        api_key_env="OPENAI_API_KEY",
    ),
    "deepseek": Provider(
        name="deepseek",
        default_model="deepseek-v4-flash",
        base_url="https://api.deepseek.com",
        api_key_env="DEEPSEEK_API_KEY",
    ),
    "openai": Provider(
        name="openai",
        default_model=None,
        base_url=None,
        api_key_env="OPENAI_API_KEY",
    ),
}


def get_provider(name: str) -> Provider:
    try:
        return _PROVIDERS[name]
    except KeyError as error:
        raise ValueError(f"Unknown provider: {name}") from error


def provider_names() -> tuple[str, ...]:
    return tuple(sorted(_PROVIDERS))

File: packages/llm/llm-openai-compatible/src/providers.ts
/** Built-in model provider presets. */

export interface Provider {
  readonly name: string;
  readonly defaultModel: string | null;
  readonly baseUrl: string | null;
  readonly suggestedModels: readonly string[];
}

const PROVIDERS: Record<string, Provider> = {
  deepseek: {
    name: "deepseek",
    defaultModel: "deepseek-v4-flash",
    baseUrl: "https://api.deepseek.com",
    suggestedModels: ["deepseek-v4-flash", "deepseek-v4-pro"],
  },
  openai: {
    name: "openai",
    defaultModel: null,
    baseUrl: null,
    suggestedModels: [],
  },
};

export function getProvider(name: string): Provider {
  const provider = PROVIDERS[name];
  if (!provider) {
    throw new Error(`Unknown provider: ${name}`);
  }
  return provider;
}

export function providerNames(): string[] {
  return Object.keys(PROVIDERS).sort();
}
